--- parameter.orig.rb	2007-11-29 22:10:31.000000000 +0100
+++ parameter.mod.rb	2007-11-29 22:11:13.000000000 +0100
@@ -87,7 +87,10 @@

         name
       end
-      @names = @names.sort.reverse
+      @names = @names.sort { |a, b|
+        size_cmp = -1 * (a.size <=> b.size) # "aaa", "aa", "a"
+        0 != size_cmp ? size_cmp : a.downcase <=> b.downcase
+      }
       @names[1..-1].each do |name|
         raise ArgumentError, "only one long name allowed (#{ @names.inspect })" if
           name.size > 1
